portal.go

```diff
@@ -630,6 +630,12 @@ func (portal *Portal) handleMatrixMessage(ctx context.Context, sender *User, evt
 	timings.convert = time.Since(start)
 	start = time.Now()
 
+	sender.Client.SendMessagesCond.L.Lock()
+	for !sender.Client.CanSendMessages {
+		sender.Client.SendMessagesCond.Wait()
+	}
+	sender.Client.SendMessagesCond.L.Unlock()
```

Comment:
This should be in the `else` block that sends unencrypted messages. Also, I don't think this alone is enough, retries are necessary too. Both the retries and this waiting should have an upper limit they'll try before giving up. The lock and bool probably shouldn't be exposed directly like this, it would be better to have a `WaitForConnection` function in `messagix.Client` that also takes a context or timeout

I'm thinking the end result should look something like this:

```go
retries := 0
for retries < 5 {
  sender.Client.WaitForConnection(15 * time.Second)
  resp, err = sender.Client.ExecuteTasks(tasks...)
  if err == nil {
    break
  }
  log.Err(err).Msg("...")
  retries++
  // Maybe also sleep here for a few seconds
}
```

Reply:
Great feedback, thank you! Done!